internal/pkg/skuba/deployments/ssh/cri.go

```diff
@@ -17,10 +17,37 @@
 
 package ssh
 
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/SUSE/skuba/pkg/skuba"
+	"github.com/pkg/errors"
+)
+
 func init() {
+	stateMap["cri.configure"] = criConfigure
 	stateMap["cri.start"] = criStart
 }
 
+func criConfigure(t *Target, data interface{}) error {
+	criFiles, err := ioutil.ReadDir(skuba.CriDir())
+	if err != nil {
+		return errors.Wrap(err, "could not read local cri directory")
+	}
+
+	defer t.ssh("rm -rf /tmp/cri.d")
+
+	for _, f := range criFiles {
+		if err := t.target.UploadFile(filepath.Join(skuba.CriDir(), f.Name()), filepath.Join("/tmp/cri.d", f.Name())); err != nil {
+			return err
+		}
+	}
+
+	_, _, err = t.ssh("cp /tmp/cri.d/crio.conf /etc/crio/crio.conf")
```

Comment:
This will replace existing cri conf. What happens if user custom tune crio before to run skuba. IIUC you want `default_capabilities` to have all the caps. Why not parse the fetch crio config file and modify?

Reply:
> What happens if user custom tune crio before to run skuba
The whole point of this PR is to enhance the **out of the box** experience, so I guess if the user is _experienced enough_ to touch the `crio.conf` they should know what they are doing. _If I would have bought a system to bootstrap k8s I would let the system do its magic_. Modifying configuration files that are essential to the core functionality of the product is not something we can do much about it -- we can document them of course,

However, I understand and value your feedback, because I also agree that skuba is doing _far more_ things than just a plain _bootstrap_ & _join_. In the ML I have suggested to Rafa to introduce an _intermediate_ command called `skuba node prepare`. This will actually take care of all the configuration _shenanigans_. I think `crio.conf` would go there as well. If you are still not persuaded, think of the same reasoning for using Terraform in the first place as it applies here as well: we are providing _one of many_ ways to do it.

> Why not parse the fetch crio config file and modify?
Ideally yes, I would prefer to parse conf and modify it. I haven't figured out how to do it yet. My idea was to modify the `/etc/sysconfig/crio` with `sed` and `regex`, but I am still not having any successful results. Although this way (parse and modify) is more risky in comparison with a plain copy-paste of a file -- I would be in favor of it because of maintenance reason. Merging the current approach would be we would have to maintain the configuration (`crio.conf`) in 2 separate places (the crio RPM package + skuba package). It goes without question if you have figured out a better way of doing that, it's more than welcome.

Reply:
> > What happens if user custom tune crio before to run skuba
> 
> The whole point of this PR is to enhance the **out of the box** experience, so I guess if the user is _experienced enough_ to touch the `crio.conf` they should know what they are doing. _If I would have bought a system to bootstrap k8s I would let the system do its magic_. Modifying configuration files that are essential to the core functionality of the product is not something we can do much about it -- we can document them of course,

We would need more than document. Think about a case when a node switching between two cluster one with and without `--strict-cap-defaults` option. That would require `skuba reset/remove` to restore `crio.conf` to its previous state. May be I am talking about some corner case here.

> It goes without question if you have figured out a better way of doing that, it's more than welcome.
Another option may be to parse the file and check if all required capabilities exist or not , when not and `--strict-cap-defaults` is set then throw error saying what skuba needs

Reply:
I've changed the PR a bit. Now instead of copy-pasting the `/etc/crio/crio.conf`, I am changing only the suse specific file, which is `/etc/sysconfig/crio`. As a result, the capabilities listed in `/etc/crio/conf` are overwritten by:

```bash
CRIO_OPTIONS=--pause-image=registry.suse.de/devel/caasp/4.0/containers/containers/caasp/v4/pause:3.1 --default-capabilities="CHOWN,DAC_OVERRIDE,FSETID,FOWNER,NET_RAW,SETGID,SETUID,SETPCAP,NET_BIND_SERVICE,SYS_CHROOT,KILL,MKNOD,AUDIT_WRITE,SETFCAP"
```

In that case, even the user modifies the `/etc/crio/crio.conf` prior to running `skuba`, their configuration will remain intact. The only thing that will be overriden will be these two things:

1. The pause container
2. the default capabilities

I will document both of those in the docs.

As for the `skuba reset` I would prefer to deal with that in a different PR, as I think there's a bug there in general. What I could do in this PR in terms of reset is:

* either remove `/etc/sysconfig/crio` and restore it by running `sudo zypper in -y -f cri-o`.
* either create a backup of the file before replacing it.

> We would need more than document. Think about a case when a node switching between two cluster one with and without --strict-cap-defaults option. That would require skuba reset/remove to restore crio.conf to its previous state. May be I am talking about some corner case here.

No. You will never end up with different container engines into nodes, simply because the configuration is created at `skuba init`. Both `node bootstrap` and `node join` are using this configuration created by the `skuba init` earlier.

Reply:
Also, I've added a `reset` functionality where it restores the `crio.conf.backup` if it exists.

Reply:
@drpaneas Looks way better now, Thanks!